packages/vscode-extension/src/webview/components/InspectDataMenu.tsx

```diff
@@ -37,40 +42,72 @@ export function InspectDataMenu({
   })();
 
   const filteredData = inspectStack.filter((item) => !item.hide);
+  const inspectItems =
+    shouldShowAll || filteredData.length === MAX_INSPECT_ITEMS + 1
+      ? filteredData
+      : filteredData.slice(0, MAX_INSPECT_ITEMS);
+  const inspectMenuAlign = inspectLocation.x <= window.innerWidth / 2 ? "start" : "end";
 
   return (
-    <div style={{ left: inspectLocation.x, top: inspectLocation.y, position: "absolute" }}>
-      <DropdownMenu.Root
-        defaultOpen={true}
-        open={true}
-        onOpenChange={(open) => {
-          if (!open) {
-            onCancel();
-          }
-        }}>
-        <DropdownMenu.Trigger />
-        <DropdownMenu.Portal>
-          <DropdownMenu.Content className="inspect-data-menu-content">
-            <DropdownMenu.Label className="inspect-data-menu-label">
-              {displayDimensionsText}
-            </DropdownMenu.Label>
-            {filteredData.map((item, index) => {
-              // extract file name from path:
-              const fileName = item.source.fileName.split("/").pop();
-              return (
-                <DropdownMenu.Item
-                  className="inspect-data-menu-item"
-                  key={index}
-                  onSelect={() => onSelected(item)}
-                  onMouseEnter={() => onHover(item)}>
-                  <code>{`<${item.componentName}>`}</code>
-                  <div className="right-slot">{`${fileName}:${item.source.line0Based + 1}`}</div>
-                </DropdownMenu.Item>
-              );
-            })}
-          </DropdownMenu.Content>
-        </DropdownMenu.Portal>
-      </DropdownMenu.Root>
-    </div>
+    <DropdownMenu.Root
+      defaultOpen={true}
+      open={true}
+      onOpenChange={(open) => {
+        if (!open) {
+          onCancel();
+        }
+      }}>
+      <DropdownMenu.Trigger asChild>
+        <span
+          style={{
+            position: "absolute",
+            left: inspectLocation.x,
+            top: inspectLocation.y,
+            opacity: 0,
+          }}></span>
+      </DropdownMenu.Trigger>
+      <DropdownMenu.Portal>
+        <DropdownMenu.Content
+          className="inspect-data-menu-content"
+          sideOffset={5}
+          align={inspectMenuAlign}
+          collisionPadding={5}>
+          <DropdownMenu.Label className="inspect-data-menu-label">
+            {displayDimensionsText}
+          </DropdownMenu.Label>
+          {inspectItems.map((item, index) => {
+            // extract file name from path:
+            const fullFileName = item.source.fileName.split("/").pop() ?? "";
+            const lastDotIndex = fullFileName.lastIndexOf(".");
+            const fileName = fullFileName.substring(0, lastDotIndex);
+            const fileExtension = fullFileName.substring(lastDotIndex);
+            return (
+              <DropdownMenu.Item
+                className="inspect-data-menu-item"
+                key={index}
```

Comment:
In general, `index` as a key is a bad practice, I think in our case when we don't have state it shouldn't cause bugs, but maybe you could use something as unique ID here? For example filePath with code line